src/main/java/com/samoilenko/onlinebookstore/repository/BookRepository.java

```diff
@@ -4,7 +4,11 @@
 import org.springframework.data.domain.Page;
 import org.springframework.data.domain.Pageable;
 import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Query;
 
 public interface BookRepository extends JpaRepository<Book, Long> {
     Page<Book> findAll(Pageable pageable);
+
+    @Query("FROM Book b LEFT JOIN FETCH b.categories c WHERE c.id = :categoryId")
```

Comment:
```suggestion
    @Query("FROM Book b LEFT JOIN b.categories c WHERE c.id = :categoryId")
```
Do you actually need to `FETCH` all of these categories, since you already know the category id?

Reply:
I assume that books can belong to several categories